### PR TITLE
ARROW-2387: [Python] Flip test for rescale loss if value < 0

### PR DESCRIPTION
**Pull request by bwo**

### CODE REVIEW

**Comment by bwo** [comment]
I see that this failed in the format check step, but when I run `make format` (with clang-format-3.9), 13 files are changed, of which only one is a file I changed. Is there a recommended version of clang-format to use?

**Comment by cpcloud** [comment]
@bwo We're using `clang-format-4.0` now, are you able to use that?

**Comment by bwo** [comment]
Yes I just saw that in the README (which is why I deleted the previous comment, not realizing it would be mirrored to jira :/). Oddly it still changed nine untouched files, but I've just committed the change to python-test.cc to see what happens.

File: cpp/src/arrow/python/python-test.cc [hunk 1]
```diff
@@ -246,6 +246,16 @@ TEST_F(DecimalTest, FromPythonDecimalRescaleTruncateable) {
...
 }
 
+TEST_F(DecimalTest, FromPythonNegativeDecimalRescale) {
+  Decimal128 value;
+  OwnedRef python_decimal(this->CreatePythonDecimal("-1.000"));
+  auto type = ::arrow::decimal(10, 9);
+  const auto& decimal_type = static_cast<const DecimalType&>(*type);
+  ASSERT_OK(
+      internal::DecimalFromPythonDecimal(python_decimal.obj(), decimal_type, &value));
+  ASSERT_EQ(-1000000000, (int64_t)(value));
```
Comment (cpcloud) [inline cpp/src/arrow/python/python-test.cc]: Don't cast here. The `operator==` definition + implicit casting rules of C++ handle that.

File: cpp/src/arrow/util/decimal.cc [hunk 2]
```diff
@@ -831,7 +831,7 @@ static bool RescaleWouldCauseDataLoss(const Decimal128& value, int32_t delta_sca
   }
 
   *result = value * multiplier;
-  return *result < value;
+  return (value < 0) ? *result < value : *result > value;
```
Comment (cpcloud) [inline cpp/src/arrow/util/decimal.cc]: I don't think this is the correct check. It should be sufficient to check that the absolute value of `*result` is less than the absolute value of `value`.

**Review by cpcloud — CHANGES_REQUESTED** [review]

File: cpp/src/arrow/util/decimal.cc (hunk 2 above)
Comment (bwo) [reply cpp/src/arrow/util/decimal.cc]: the check was in fact backward (for entirely stupid reasons). Do you want me to use an actual absolute value method or is it fine now?

**Comment by cpcloud** [comment]
@bwo Looks like this is failing for unrelated reasons, can you rebase on top of master and push again? Then we can merge.

**Review by cpcloud — APPROVED** [review]

**Comment by bwo** [comment]
Seems to be failing on something Ruby-related? Should I rebase again?

**Comment by cpcloud** [comment]
@bwo Yes.

**Comment by bwo** [comment]
hooray!

**Comment by cpcloud** [comment]
Sweet! Merging. Thank you!

**Pull request closed without merging**